paddle/fluid/inference/api/analysis_predictor.cc

```diff
@@ -238,6 +247,11 @@ bool AnalysisPredictor::Run(const std::vector<PaddleTensor> &inputs,
   // recover the cpu_math_library_num_threads to 1, in order to avoid thread
   // conflict when integrating it into deployment service.
   paddle::platform::SetNumThreads(1);
+#ifdef PADDLE_WITH_MKLDNN
+  // TODO(intel): will refactor this code later
+  // To avoid confusion when thread is reused from pool
+  if (config_.mkldnn_disable_cache_ > 0) paddle::platform::set_cur_thread_id(0);
```

Comment:
cur_thread_id is by default 0 , why to set it to 0 when disabling caching is requested ? Shouldn't it be set to non-zero value , so it match changes in Set_Blob (device_context.cc:417)

Reply:
it is to clear the setting before exit execution from thread, and setting is stored in thread_local variable. The thread is reused from pool, if not clean, it may impact others which don't need disabling cache.

Reply:
Ok, got it.